static/src/javascripts/projects/common/modules/identity/sign-in-gate/helper.js

```diff
@@ -253,6 +253,22 @@ export const isInvalidSection = (include: Array<string> = []): boolean => {
         }, false);
 };
 
+// hide the sign in gate for certain tags on the site
+// add to the include parameter array if there are specific tags that should be included/overridden
+export const isInvalidTag = (include: Array<String> = []): boolean => {
+    const invalidTags = [
+        'newsletters/newsletters'
+    ];
+
+    return invalidTags
+        .filter(el => !include.includes(el))
+        .reduce((isTagInvalid, tag) => {
+            if (isTagInvalid) return true;
+            // look up window.guardian.config object in browser console
+            return config.get(`page.keywordIds`, "").split(',').includes(tag)
+        }, false);
```

Comment:
I wonder if this is a better optimisation rather than the reduce, as I think it's equivalent?

Although we didn't implement `some` on frontend in the other examples (`isInvalidArticleType` and `isInvalidSection`), we do a similar thing on [DCR](https://github.com/guardian/dotcom-rendering/blob/46d4a9f1e15b722e94d66c91d3f79fa1bf18ccf2/src/web/components/SignInGate/displayRule.ts#L62)

But I suppose it's also fine to leave it as `reduce`. 🤔

```suggestion
        .some((tag) => config.get(`page.keywordIds`, "").split(',').includes(tag));
```

Reply:
I much prefer `some` here for both brevity and the fact that it's clearer to understand intent than when using `reduce`